qiskit_nature/circuit/library/ansaetze/ucc.py

```diff
@@ -0,0 +1,213 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+TODO.
+"""
+
+from functools import partial
+from typing import Callable, List, Optional, Tuple, Union
+
+import logging
+
+from qiskit_nature import QiskitNatureError
+from qiskit_nature.operators.second_quantization import SecondQuantizedOp
+from qiskit_nature.operators.second_quantization.qubit_converter import QubitConverter
+from .evolved_operator_ansatz import EvolvedOperatorAnsatz
+from .excitation_builder import ExcitationBuilder
+
+logger = logging.getLogger(__name__)
+
+
+class UCC(EvolvedOperatorAnsatz):
+    """The Unitary Coupled-Cluster Ansatz."""
```

Comment:
It would be nice to have some examples on how to call this class in the docstring